lnrpc/marshall_utils.go

```diff
@@ -179,3 +180,67 @@ func MarshalOutPoint(op *wire.OutPoint) *OutPoint {
 		OutputIndex: op.Index,
 	}
 }
+
+// ParseCoinSelectionStrategyObject parses a wallet.CoinSelectionStrategy
+// object to its string counterpart.
+func ParseCoinSelectionStrategyObject(strategy wallet.CoinSelectionStrategy) (
+	*string, error) {
+
+	switch strategy {
+	case wallet.CoinSelectionLargest:
+		coinSelectionStrategy := "largest"
+		return &coinSelectionStrategy, nil
+	case wallet.CoinSelectionRandom:
+		coinSelectionStrategy := "random"
+		return &coinSelectionStrategy, nil
+	default:
+		return nil, fmt.Errorf("unknown coin selection strategy "+
+			"%v", strategy)
+	}
+}
+
+// ParseCoinSelectionStrategy parses a coin selection strategy string
+// from the CLI to its lnrpc.CoinSelectionStrategy counterpart proto type.
+func ParseCoinSelectionStrategy(strategy string) (
+	*CoinSelectionStrategy, error) {
+
+	switch strategy {
+	case "global-config":
+		strategyVal := CoinSelectionStrategy_STRATEGY_USE_GLOBAL_CONFIG
+		return &strategyVal, nil
+	case "largest":
+		strategyVal := CoinSelectionStrategy_STRATEGY_LARGEST
+		return &strategyVal, nil
+	case "random":
+		strategyVal := CoinSelectionStrategy_STRATEGY_RANDOM
+		return &strategyVal, nil
+	default:
+		return nil, fmt.Errorf("unknown coin selection strategy "+
+			"%v", strategy)
+	}
+}
+
+// UnmarshallCoinSelectionStrategy converts a lnrpc.CoinSelectionStrategy proto
+// type to its wallet.CoinSelectionStrategy counterpart type.
+func UnmarshallCoinSelectionStrategy(strategy CoinSelectionStrategy,
+	globalStrategy string) (wallet.CoinSelectionStrategy, error) {
+
+	switch strategy {
+	case CoinSelectionStrategy_STRATEGY_USE_GLOBAL_CONFIG:
+		strategyValue, _ := ParseCoinSelectionStrategy(globalStrategy)
+		coinSelectionStrategy, _ := UnmarshallCoinSelectionStrategy(
```

Comment:
ok I see now that this wont actually loop but this is purely because we know that the string `globalStrategy` will be one of "random" or "largest". 

I think we can instead be a bit more strict about the structure here instead though:

[Here](https://github.com/lightningnetwork/lnd/blob/5cb4811e86ed1607c1c9671756d7276256e2ec44/config_builder.go#L590) is where we parse the "coin-selection-strategy" config option. We should only ever do that conversion from string to type once. Then, we can instead just return that wallet.CoinSelectionStrategy type here